resources/dynamodb-items.go

```diff
@@ -0,0 +1,81 @@
+package resources
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+type DynamoDBTableItem struct {
+	svc   *dynamodb.DynamoDB
+	id    map[string]*dynamodb.AttributeValue
+	table Resource
+}
+
+func (n *DynamoDBNuke) ListItems() ([]Resource, error) {
+	tables, tablesErr := n.ListTables()
+	if tablesErr != nil {
+		return nil, tablesErr
+	}
+
+	resources := make([]Resource, 0)
+	for _, dynamoTable := range tables {
+		describeParams := &dynamodb.DescribeTableInput{
+			TableName: aws.String(dynamoTable.String()),
+		}
+
+		descResp, descErr := n.Service.DescribeTable(describeParams)
+		if descErr != nil {
+			return nil, descErr
+		}
+
+		key := *descResp.Table.KeySchema[0].AttributeName
+		params := &dynamodb.ScanInput{
+			TableName:            aws.String(dynamoTable.String()),
+			ProjectionExpression: aws.String(key),
+		}
+
+		scanResp, scanErr := n.Service.Scan(params)
+		if scanErr != nil {
+			return nil, scanErr
+		}
+
+		for _, itemMap := range scanResp.Items {
+			resources = append(resources, &DynamoDBTableItem{
+				svc:   n.Service,
+				id:    itemMap,
+				table: dynamoTable,
+			})
+		}
+	}
+
+	return resources, nil
+}
+
+func (i *DynamoDBTableItem) Remove() error {
+	params := &dynamodb.DeleteItemInput{
+		Key:       i.id,
+		TableName: aws.String(i.table.String()),
+	}
+
+	_, err := i.svc.DeleteItem(params)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (i *DynamoDBTableItem) String() string {
+	table := i.table.String()
+	keyField := ""
+
+	for _, value := range i.id {
+		value := strings.Replace(value.String(), "\n", " ", -1)
+		keyFieldChars := []rune(value)
+		keyField = string(keyFieldChars[8:(len(keyFieldChars) - 3)])
```

Comment:
Slicing around with strings really looks hacky. Also it's hard to understand, since I have no idea how the contents look like.

So why do you need to do that?

Does this help? https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/dynamodbattribute/#ConvertFrom